### ISSUE TITLE
Want an updated Makefile for BSD/Homebrew/non-root?

### ISSUE THREAD

**Issue by apjanke** [tj/git-extras#463]
Hi folks,

I've just discovered git-extras through my work with Oh My Zsh and Homebrew and I'm digging it; I'd like to start using it regularly. I'm mainly an OS X/Homebrew user with a multi-user, multi-machine setup, and I'm running in to a few issues on this platform. I've got space for a weekend project. Would you be interested in a PR that refactored the `Makefile` and build process to do some or all of:

* Make sure the whole build, including doco generation, works under BSD/OS X
* Support relocating `etc` (#237)
* Support disabling selected commands and suppressing prompting for alias conflicts (#442, #228)
 * (will make fully unattended, repeatable installs possible)
 * Remove man & doco for disabled commands, just to be tidy
* Split the build and install into two steps, so you could build the commands (slurping in the library files) and run them inside the local source tree without having to install to the final destination (useful for iterating development)
* Install the HTML and other format docs under `/usr/share/doc` (right now it looks like they're going under `man`, which should really only be for man-formatted stuff)
* Unify `makefile.msys` with the main `Makefile` (#427)

Regardless, cheers for the good work.

**Comment by spacewander** [comment]
Waiting for it :smile:

**Comment by qw3rtman** [comment]
My only concern is that the build and install into two steps should be opt-in and not opt-out, so that the installation process doesn't become more complicated.

Otherwise, looking forward to seeing a PR from you. Thanks! :+1:

**Comment by apjanke** [comment]
Yeah, it'll be opt-in and transparent to users: `make install` will do the same thing as it does now, and do the `build` step internally as a dependency. This will just give developers the option to request just a local build with a `make build` or similar command.

**Comment by qw3rtman** [comment]
Sounds like a good idea to me! :smile:

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Hyderabad go to  
http://www.expert5th.in/packers-and-movers-hyderabad/
For Expert 5 Packers and Movers Bangalore go to 
http://www.expert5th.in/packers-and-movers-bangalore/
For Expert 5 Packers and Movers Mumbai go to 
http://www.expert5th.in/packers-and-movers-mumbai/

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Gurgaon go to 
http://www.expert5th.in/packers-and-movers-gurgaon/
For Expert 5 Packers and Movers Delhi go to 
http://www.expert5th.in/packers-and-movers-delhi/
For Expert 5 Packers and Movers Pune go to  
http://www.expert5th.in/packers-and-movers-pune/

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Chennai go to 
http://www.expert5th.in/packers-and-movers-chennai/
For Expert 5 Packers and Movers Noida go to 
http://www.expert5th.in/packers-and-movers-noida/
For Expert 5 Packers and Movers Thane go to 
http://www.expert5th.in/packers-and-movers-thane/
For Expert 5 Packers and Movers Navi Mumbai go to 
http://www.expert5th.in/packers-and-movers-navimumbai/
For Expert 5 Packers and Movers Ghaziabad go to 
http://www.expert5th.in/packers-and-movers-ghaziabad/
For Expert 5 Packers and Movers Faridabad go to 
http://www.expert5th.in/packers-and-movers-faridabad/

**Comment by raviranjananalist** [comment]
For Expert 5 Packers and Movers Bangalore go to 
http://www.expert5th.in/packers-and-movers-bangalore/
For Expert 5 Packers and Movers Pune go to  
http://www.expert5th.in/packers-and-movers-pune/
For Expert 5 Packers and Movers Mumbai go to 
http://www.expert5th.in/packers-and-movers-mumbai/
